File: javascript/random-number-generator/random-number-generator.js
// Random number generator in JavaScript
// This script generates a random number based on the current time in milliseconds.
// Refactored to output to console.
// Usage: node random-number-generator.js

function randomNumberGenerator() {
  const now = new Date();
  const timeInMs = now.getTime(); // current time in milliseconds
  const randomNumber = Math.floor(timeInMs % 1000); // Take the remainder of the division of the current time in milliseconds by 1000
  return randomNumber;
}

const randomNumber = randomNumberGenerator();
console.log(randomNumber);

File: python/random-number-generator/random-number-generator.py
# Generates a random number based on time

import time


def random_number_generator():
    """
    Generates a random number based on the current time in milliseconds.
    """
    now = time.time()
    random_number = int(now * 1000) % 1000
    return random_number


random_num = random_number_generator()
print(random_num)
